pkg/agent/openflow/client_test.go

```diff
@@ -152,58 +150,33 @@ func TestConcurrentFlowInstallation(t *testing.T) {
 			client := ofClient.(*client)
 			client.flowOperations = m
 
-			var wg sync.WaitGroup
-			var counter int32        // tracks the number of concurrent Add calls at any given time
-			var callIdx int32        // used to distinguish between the first Add call and subsequent calls
-			var concurrentCalls bool // set to true if we observe concurrent calls
-			var stop bool
-			var mutex sync.Mutex
-			cond := sync.NewCond(&mutex)
+			var concurrentCalls atomic.Bool // set to true if we observe concurrent calls
 
+			addCh := make(chan int)
 			m.EXPECT().Add(gomock.Any()).DoAndReturn(func(args ...interface{}) error {
-				cond.L.Lock()
-				defer cond.L.Unlock()
-				counter += 1
-				callIdx += 1
-				if callIdx == 1 { // first call
-					// we wait until we need to stop
-					// we stop when one of these conditions is met:
-					// 1) we detected concurrent Add calls (test is a success), or
-					// 2) the test timed out
-					for !stop {
-						cond.Wait()
-					}
-				} else if counter > 1 {
-					concurrentCalls = true
-					stop = true
-					cond.Signal()
+				x := rand.Int()
```

Comment:
there is technically a possibility that 2 consecutive calls to `rand.Int()` can return the same value, so maybe use 2 different static values?

Reply:
A second thought, we actually don't need to care data in the channel, the `select` guarantees only one case would be executed.

Reply:
I changed the random number to UUID.